src/add-candidate.php

```diff
@@ -514,6 +601,118 @@ function duplicateForm() {
+                    $('.cancel-button').on('click', function(e) {
+                        if (isDirty) {
+                            e.preventDefault();
+                            $('#warningModal').modal('show');
+                        } else {
+                            window.location.href = 'add-candidate.php';
+                        }
+                    });
+
+                    // Handle leave button click in the modal
+                    $('#leaveButton').on('click', function() {
+                        isDirty = false;
+                        window.removeEventListener('beforeunload', showWarningModal);
+                        window.location.href = targetUrl;
+                    });
+
+                    $('.modal .cancel').on('click', function() {
+                        $('#warningModal').modal('hide');
+                    });
+
+                    function showWarningModal(e) {
+                        if (isDirty) {
+                            e.preventDefault();
+                            $('#warningModal').modal('show');
+                            return ''; // Required for some browsers to show the modal
+                        }
+                    }
+
+                    window.addEventListener('beforeunload', showWarningModal);
+
+                    $(document).on('click', 'a', function(e) {
+                        if (isDirty) {
+                            e.preventDefault();
+                            targetUrl = $(this).attr('href');
+                            $('#warningModal').modal('show');
+                        }
+                    });
+                });
+                //reset form modal
+                document.addEventListener("DOMContentLoaded", function() {
+                    let isFormDirty = false;
+                    const form = document.querySelector("form");
+                    const resetButton = document.querySelector("button[type='reset']");
+                    const resetModal = new bootstrap.Modal(document.getElementById('resetFormModal'));
+
+                    // Mark the form as dirty if any input changes
+                    form.addEventListener("input", function() {
+                        isFormDirty = true;
+                    });
+
+                    // Show the reset modal on reset button click
+                    resetButton.addEventListener("click", function(event) {
+                        if (isFormDirty) {
+                            event.preventDefault();
+                            resetModal.show();
+                        }
+                    });
+
+                    // Handle form reset confirmation
+                    document.getElementById("confirmReset").addEventListener("click", function() {
+                        isFormDirty = false;
+                        form.reset();
+                        resetModal.hide();
+                    });
+                });
+            </script>
```

Comment:
Put this in a separate or existing .js file

Reply:
fixed this with new file name: candidate-modal.js